### ISSUE TITLE
NuGet export such that Visual Studio copies DLLs to output directory?

### ISSUE THREAD

**Issue by adv12** [microsoft/vcpkg#20272]
I'm a C++ newbie trying to use a NuGet package exported from vcpkg in an existing Visual C++ 2019 project.  I'm able to successfully build my project against the installed NuGet package, but I had the expectation that DLLs from the NuGet package's bin directory would get copied to my project's output directory, and they don't.

Is this something I can change about how I'm consuming the NuGet package, or would this behavior require a change to the NuGet package itself?  I'm kind of wondering how anyone can actually use these exported NuGet packages without the behavior I was expecting.

**Comment by misirlou-tg** [comment]
What package are you importing into your existing Visual C++ 2019 project? Are you importing it with NuGet or vcpkg?

**Comment by adv12** [comment]
As stated in my question, I'm importing a NuGet package exported from vcpkg.  That NuGet package is exported with a command line that looks something like this:

vcpkg export gdal expat libkml minizip[bzip2] uriparser boost-smart-ptr xerces-c sqlite3 --overlay-ports=<my overlay ports directory> --nuget --output=<my NuGet package name>

I have added the NuGet package produced by this command to a local NuGet package repository, added that repository to my list of Package Sources in Visual Studio, browsed that repo in Visual Studio, and installed the NuGet package for the existing VC++ project in question.  So the VC++ project is now looking to the NuGet package for .h and .lib files.

The existing VC++ project, which references some of the libraries in the NuGet packages, builds successfully, so I am persuaded that I have correctly exported and installed the NuGet package.

The one thing I was expecting that doesn't happen is the .dll files from the NuGet package getting copied to the output directory of my VC++ project on successful build.

**Comment by adv12** [comment]
I've poked around in the expanded NuGet package and found the .targets file with the AppLocalFromInstalled target that performs this copy.  Running msbuild from the command line, I see that it is getting run, and I've found one project in my solution where .dlls from the NuGet package did, indeed, get copied to the output directory.  I actually have multiple projects in the solution linking to libs in this NuGet package, and I don't know why, in some of them, I'm not seeing the DLLs get copied up.  But given that it does seem to be happening in some cases, I'll assume this is user error and close this issue.
